packages/core/base/src/standard.ts

```diff
@@ -0,0 +1,33 @@
+import type {
+    SolanaSignAndSendTransactionFeature,
+    SolanaSignMessageFeature,
+    SolanaSignTransactionFeature,
+} from '@solana/wallet-standard-features';
+import type { Wallet as StandardWallet, WalletWithFeatures as StandardWalletWithFeatures } from '@wallet-standard/base';
+import type { ConnectFeature, DisconnectFeature, EventsFeature } from '@wallet-standard/features';
+import type { WalletAdapter, WalletAdapterProps } from './adapter.js';
+
+export type WalletAdapterCompatibleStandardWallet = StandardWalletWithFeatures<
+    ConnectFeature &
+        EventsFeature &
+        (SolanaSignAndSendTransactionFeature | SolanaSignTransactionFeature) &
+        (DisconnectFeature | SolanaSignMessageFeature | never)
+>;
+
+export interface StandardWalletAdapterProps<Name extends string = string> extends WalletAdapterProps<Name> {
+    wallet: WalletAdapterCompatibleStandardWallet;
+    standard: true;
+}
+
+export type StandardWalletAdapter<Name extends string = string> = WalletAdapter<Name> &
+    StandardWalletAdapterProps<Name>;
+
+export function isWalletAdapterCompatibleStandardWallet(
+    wallet: StandardWallet
+): wallet is WalletAdapterCompatibleStandardWallet {
+    return (
+        'standard:connect' in wallet.features &&
+        'standard:events' in wallet.features &&
+        ('solana:signAndSendTransaction' in wallet.features || 'solana:signTransaction' in wallet.features)
+    );
```

Comment:
Does this function need an equivalent of the line `(DisconnectFeature | SolanaSignMessageFeature | never)` from the `WalletAdapterCompatibleStandardWallet` type?

Reply:
No, because the set of possible features on the type is larger than the required features (i.e., they are optional). We don't check for them at runtime _here_ to determine if a Standard Wallet is Wallet Adapter-compatible. However, we do later look for these optional features: https://github.com/solana-labs/wallet-standard/blob/a4d058a11caac4d9cd79fcdef39dd7d31656fb18/packages/wallet-adapter/base/src/adapter.ts#L432-L439